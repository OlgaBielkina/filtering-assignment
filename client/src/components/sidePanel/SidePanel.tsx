import React, { useState } from 'react';
import classnames from 'classnames';
import styles from './SidePanel.module.scss';

interface Props {
    text: string;
    children: React.ReactNode;
}

export const SidePanel: React.FC<Props> = ({ text, children }) => {
    const [panelOpened, setPanelOpened] = useState<boolean>(false);
    const toggleFilters = () => {
        setPanelOpened(!panelOpened);
    };

    return (
        <div>
            <button className={styles.toggleButton} onClick={toggleFilters}>
                {text}
            </button>
            <aside data-testid="side-panel" className={classnames(styles.panel, panelOpened ? styles.visible : '')}>
                <span onClick={toggleFilters} className={classnames('material-icons', styles.closeButton)}>
                    close
                </span>
                {children}
            </aside>
        </div>
    );
};
