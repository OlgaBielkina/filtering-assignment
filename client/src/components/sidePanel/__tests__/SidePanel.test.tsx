import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { SidePanel } from '../SidePanel';

describe('SidePanel', () => {
    const buttonText = 'toggle panel';

    it('should render mobile side panel', () => {
        global.innerWidth = 760;
        window.dispatchEvent(new Event('resize'));

        const { asFragment } = render(
            <SidePanel text={buttonText}>
                <span></span>
            </SidePanel>,
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('should open side panel', () => {
        global.innerWidth = 760;
        window.dispatchEvent(new Event('resize'));

        const { getByTestId, getByText } = render(
            <SidePanel text={buttonText}>
                <span></span>
            </SidePanel>,
        );
        expect(getByTestId('side-panel')).not.toHaveClass('visible');

        fireEvent.click(getByText(buttonText));

        expect(getByTestId('side-panel')).toHaveClass('visible');
    });

    it('should close side panel', () => {
        global.innerWidth = 760;
        window.dispatchEvent(new Event('resize'));

        const { getByTestId, getByText } = render(
            <SidePanel text={buttonText}>
                <span></span>
            </SidePanel>,
        );
        expect(getByTestId('side-panel')).not.toHaveClass('visible');

        fireEvent.click(getByText(buttonText));

        expect(getByTestId('side-panel')).toHaveClass('visible');

        fireEvent.click(getByText('close'));

        expect(getByTestId('side-panel')).not.toHaveClass('visible');
    });
});
