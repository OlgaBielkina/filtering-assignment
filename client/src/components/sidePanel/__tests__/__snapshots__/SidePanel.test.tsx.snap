// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel should render mobile side panel 1`] = `
<DocumentFragment>
  <div>
    <button
      class="toggleButton"
    >
      toggle panel
    </button>
    <aside
      class="panel"
      data-testid="side-panel"
    >
      <span
        class="material-icons closeButton"
      >
        close
      </span>
      <span />
    </aside>
  </div>
</DocumentFragment>
`;
